### PR TITLE
IDEMPIERE-5259 Labels feature

### PR DESCRIPTION
**Pull request by alanlesc1**

### CODE REVIEW

**Comment by hengsin** [comment]
Please add javadoc for all public constructor and methods (except "@Override" methods).

For theme related changes, you need to update the "idempiere.theme.default" version timestamp in org.adempiere.ui.zk/WEB-INF/src/metainfo/zk/lang-addon.xml

**Comment by alanlesc1** [comment]
Thanks @hengsin. Changes committed.

File: org.adempiere.base/src/org/compiere/model/MSysConfig.java [hunk 1]
```diff
@@ -231,6 +231,7 @@ public class MSysConfig extends X_AD_SysConfig
     public static final String ZK_TOOLBAR_SHOW_MORE_VERTICAL = "ZK_TOOLBAR_SHOW_MORE_VERTICAL";
     public static final String ZK_USE_PDF_JS_VIEWER = "ZK_USE_PDF_JS_VIEWER";
     public static final String ZOOM_ACROSS_QUERY_TIMEOUT = "ZOOM_ACROSS_QUERY_TIMEOUT";
+    public static final String LABEL_AUTOMATIC_COLOR = "LABEL_AUTOMATIC_COLOR";
```
Comment (CarlosRuiz-globalqss) [inline org.adempiere.base/src/org/compiere/model/MSysConfig.java:234]: MSysConfig is ordered alphabetically

Comment (alanlesc1) [reply org.adempiere.base/src/org/compiere/model/MSysConfig.java]: thanks. Changes commited

**Review by CarlosRuiz-globalqss — APPROVED** [review]
tests in https://test-feature.idempiere.org:3243/webui/ are OK

**Comment by norbertbede** [comment]
hi @CarlosRuiz-globalqss @alanlesc1 

let me ask about the status of this PR. I reported some issues with speed of entry.
we would like implement this feature, our system.
thanks.
Norbert

**Comment by hengsin** [comment]
hi @alanlesc1, please resolve the merge conflict for org.adempiere.ui.zk/WEB-INF/src/metainfo/zk/lang-addon.xml

**Comment by alanlesc1** [comment]
> hi @alanlesc1, please resolve the merge conflict for org.adempiere.ui.zk/WEB-INF/src/metainfo/zk/lang-addon.xml

Done! thanks @hengsin

**Comment by CarlosRuiz-globalqss** [comment]
> hi @CarlosRuiz-globalqss @alanlesc1
> 
> let me ask about the status of this PR. I reported some issues with speed of entry. we would like implement this feature, our system.


Hi @norbertbede 

I installed it again in https://test-feature.idempiere.org:3243/webui/ in case you want to conduct further tests.

I think this is already tested by @hengsin and me, so is good to go when you find the performance and duplicates issue solved.

**Comment by norbertbede** [comment]
Thanks @alanlesc1  for the fix. 
@CarlosRuiz-globalqss tested, PASS. (duplicates are not created, performance since slower, but works as we need.)
nb

**Pull request merged**